cmd/thor/utils.go

```diff
@@ -54,18 +56,44 @@ import (
 	"github.com/vechain/thor/v2/tx"
 	"github.com/vechain/thor/v2/txpool"
 	"gopkg.in/urfave/cli.v1"
-
-	ethlog "github.com/ethereum/go-ethereum/log"
 )
 
 var devNetGenesisID = genesis.NewDevnet().ID()
 
-func initLogger(lvl log15.Lvl) {
-	log15.Root().SetHandler(log15.LvlFilterHandler(lvl, log15.StderrHandler))
-	// set go-ethereum log lvl to Warn
-	ethLogHandler := ethlog.NewGlogHandler(ethlog.StreamHandler(os.Stderr, ethlog.TerminalFormat(true)))
-	ethLogHandler.Verbosity(ethlog.LvlWarn)
-	ethlog.Root().SetHandler(ethLogHandler)
+func initLogger(ctx *cli.Context) {
+	logLevel := log.FromLegacyLevel(ctx.Int(verbosityFlag.Name))
+	jsonLogs := ctx.Bool(jsonLogsFlag.Name)
+	output := io.Writer(os.Stdout)
+
+	var handler slog.Handler
+	if jsonLogs {
+		handler = log.JSONHandlerWithLevel(output, logLevel)
+	} else {
+		useColor := (isatty.IsTerminal(os.Stderr.Fd()) || isatty.IsCygwinTerminal(os.Stderr.Fd())) && os.Getenv("TERM") != "dumb"
+		handler = log.NewTerminalHandlerWithLevel(output, logLevel, useColor)
+	}
+	log.SetDefault(log.NewLogger(handler))
+	ethlog.Root().SetHandler(&ethLogger{
+		logger: log.WithContext("pkg", "geth"),
+	})
+}
+
+type ethLogger struct {
+	logger log.Logger
+}
+
+func (h *ethLogger) Log(r *ethlog.Record) error {
+	switch r.Lvl {
+	case ethlog.LvlCrit:
+		h.logger.Crit(r.Msg)
+	case ethlog.LvlError:
+		h.logger.Error(r.Msg)
+	case ethlog.LvlWarn:
+		h.logger.Warn(r.Msg)
+	default:
+		return nil
```

Comment:
Is this missing a few levels like debug and info ?

Reply:
Previously we weren't logging anything below warn, so I just didn't bother implementing them